### PR TITLE
tests, net: Replace semi-dynamic IPv6 configuration with a static one

### PR DESCRIPTION
**Pull request by EdDev**
**What this PR does / why we need it**:

The bridge and masquerade network bindings do not support dynamic IPv6 allocation, or more accurately, it supports only DHCPv6 without RA.

In practice, it requires users to define IPv6 routes manually. Tests have been using this method to configure IPv6:
```
dhclient -6 eth0
ip -6 route add <Network CIDR> dev eth0
ip -6 route add default via <gateway>
```

In addition, cloud-init network configuration have been setting DHCPv6 as enabled, causing the guest client (in the tests, it is NetworkManager) to expect IPv6 dynamic allocation. While older NM version did not block on not receiving an address, new ones do.

Therefore, this change replaces the cloud-init network DHCPv6 setting to a static IPv6 configuration and removes the manual configuration in the guest.
There is no practical need to support this semi-dynamic IPv6 configuration.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by EdDev** [comment]
> I believe [this](https://github.com/kubevirt/kubevirt/blob/8c6ef993a80b7f6a8fdd9a5caffa73bbe6e8c736/tests/network/vmi_networking.go#L959) can be removed as well.

Right, the static settings should not have changed.

> However, I think we should keep one test that is testing DHCPv6. Or, another option. remove DHCPv6 from the code base. I don't see a point of keeping it in the code base without any coverage.

I would not touch it as we may want to complete the support.
But I do not mind either way.

I guess we can leave the current `dhclient` you linked above as is to see it works.

File: tests/network/vmi_networking.go [hunk 1]
```diff
@@ -653,10 +653,14 @@ var _ = SIGDescribe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:c
...
 
 		fedoraMasqueradeVMI := func(ports []v1.Port, ipv6NetworkCIDR string) (*v1.VirtualMachineInstance, error) {
+			if ipv6NetworkCIDR == "" {
+				ipv6NetworkCIDR = libnet.DefaultIPv6CIDR
+			}
 			networkData, err := libnet.NewNetworkData(
 				libnet.WithEthernet("eth0",
 					libnet.WithDHCP4Enabled(),
-					libnet.WithDHCP6Enabled(),
+					libnet.WithAddresses(ipv6NetworkCIDR),
```
Comment (AlonaKaplan) [inline tests/network/vmi_networking.go:662]: The test fails since the CIDR is used here instead of the address.

Comment (EdDev) [reply tests/network/vmi_networking.go:662]: I pushed a [fix](https://github.com/kubevirt/kubevirt/compare/31fe1457f726ed382cb76043022571d1f9d7dbb8..b479dd634471661a111f88dcb7a7395c16848d7b), hopefully.

**Comment by AlonaKaplan** [comment]
> I guess we can leave the current dhclient you linked above as is to see it works.

I think in this case it is just redundant. To test DHCP6 you need to run the VM without specifying the IP via the cloudinit.

**Comment by AlonaKaplan** [comment]
Now we don't have any e2e testing DHCP6.
But it can be added in a follow-up.

/approve

**Pull request merged**